providers/com/peerj/normalizer.py

```diff
@@ -0,0 +1,160 @@
+import arrow
+
+import dateparser
+
+from share.normalize import Parser, Static, Delegate, RunPython, ParseName, Normalizer, Maybe, Concat, Map, ctx, Try
+from share.normalize.utils import format_doi_as_url
+
+
+class ISSN(Parser):
+    schema = 'Link'
+    url = ctx
+    type = Static('issn')
+
+
+class Email(Parser):
+    email = ctx
+
+
+class PersonEmail(Parser):
+    email = Delegate(Email, ctx)
+
+
+class Institution:
+    name = ctx.author_institution
+
+
+class Affiliation:
+    # The entity used here could be any of the entity subclasses (Institution, Publisher, Funder, Organization).
+    entity = Delegate(Institution, ctx)
+
+
+class Link(Parser):
+    url = ctx
+    type = RunPython('get_link_type', ctx)
+
+    def get_link_type(self, link):
+        if 'dx.doi.org' in link:
+            return 'doi'
+        if 'peerj.com' in link:
+            return 'provider'
+        return 'misc'
+
+
+class ThroughLinks(Parser):
+    link = Delegate(Link, ctx)
+
+
+class Tag(Parser):
+    name = ctx
+
+
+class ThroughTags(Parser):
+    tag = (Delegate(Tag), ctx)
+
+
+class Publisher(Parser):
+    name = ctx
+
+
+class Institution(Parser):
+    name = RunPython('get_author_institute', ctx)
+
+    def get_author_institute(self, context):
```

Comment:
Umm, so it auto changed context to the full dictionary in the `get_author_institution` function, but I assume you meant see what happens if `institutions` in not a value in the dictionary.